### DISCUSSION TITLE
is it fine to use 2 server ports in one uwrbsockets.js server?

### DISCUSSION THREAD

**Discussion by Dhruv-Garg79** [uNetworking/uWebSockets.js#974] [General]
I want to use 2 server ports:
- 3000 for public APIs
- 4000 for internal APIs

I will expose public APIs using a public load balancer and internal APIs using a private load balancer. 

I wanted to know is listening to 2 ports on the same server a good idea?

**Comment by uNetworkingAB** [comment]
Yes you just call .listen twice that's totally fine

**Reply by Dhruv-Garg79** [reply]
I am just confused about one thing.

According to my simple understanding of a TCP/HTTP server, we listen for a socket connection on a port in a while loop. If we listen for 2 ports we need to check if there is a request at any of these ports.

so does that have any adverse performance impact?

p.s. I know I might be wrong, I don't have good enough knowledge of this.

**Reply by uNetworkingAB** [reply]
The event loop that drives your server app calls epoll_wait in a loop. It doesn't matter if you have 1 or 50 listen sockets, you still only call epoll_wait once and it returns when there are 1 or more events, regardless of how many you listen to

**Reply by Dhruv-Garg79** [reply]
understood. can you please tell me some resources for understanding the internal workings better?

**Comment by uasan** [comment]
All your routes will be available on 2 ports, i.e. There is no privacy at the routing level

**Reply by Dhruv-Garg79** [reply]
got it.
